[make:auth] Add `RememberMeBadge`

bechir opened the pull request:
Fix #848 

@weaverryan in #848:
> EDIT: And I wonder if we should even ask "Do you want to support remember me?" during this process? We could then ask "Do you want remember me to be activated via a checkbox or always activated"? We could use this to determine how the template is generated AND to automatically add the correct remember_me config to security.yaml.


- [x] Ask "Do you want to support remember me?" and "Do you want remember me to be activated via a checkbox or always activated"?
- [x] Add RememberMeBadge in `LoginFormAuthenticator` when remember me is supported
- [x] Update generated `security.yaml`
- [x] Update `security/login.html.twig`

weaverryan commented on src/Maker/MakeAuthenticator.php:
```diff
@@ -194,20 +194,43 @@ function ($answer) {
+                    $io->confirm(
+                        'Do you want remember me to be activated via a checkbox or always activated?',
+                        false
+                    )
```
Hmm, the question isn't very clear. Perhaps the user should choose between 2 options instead of a yes/no:

[0] Activate when the user checks a box
[1] Always activate remember me

weaverryan commented on src/Resources/skeleton/authenticator/Security52LoginFormAuthenticator.tpl.php:
```diff
@@ -39,7 +39,7 @@ public function authenticate(Request $request): PassportInterface
             new PasswordCredentials($request->request->get('password', '')),
             [
-                new CsrfTokenBadge('authenticate', $request->request->get('_csrf_token')),
+                new CsrfTokenBadge('authenticate', $request->request->get('_csrf_token')),<?= $remember_me_badge ? "\n\t\t\t\tnew RememberMeBadge(),\n" : "\n" ?>
```
That's ugly, but I can't remember if there is a nicer way to do it :/

weaverryan commented on src/Resources/skeleton/authenticator/login_form.tpl.php:
```diff
@@ -25,6 +25,29 @@
+                <input type="checkbox" name="_remember_me"> Remember me
+            </label>
+        </div>
+    #}
```
I don't think we need this first section at all: if they choose "always remember me", let's not generate anything in this template.

weaverryan commented on src/Resources/skeleton/authenticator/login_form.tpl.php:
```diff
@@ -36,6 +59,7 @@
             </label>
         </div>
     #}
```
Now that we're asking the question explicitly, if the user chooses "no" to remember me, let's not generate anything (so delete the `else` case and the code above)

weaverryan commented on src/Security/SecurityConfigUpdater.php:
```diff
@@ -138,6 +138,30 @@ public function updateForAuthenticator(string $yamlSource, string $firewallName,
+            } else {
+                $firewall['remember_me'][] = $this->manipulator->createCommentLine('always_remember_me: true');
+            }
+        }
```
Wow, you've done an amazing job being VERY user-friendly here :)

bechir commented:
Hi @weaverryan !

Thank you for your feedbacks.

Sorry for the late. I just saw that I had a PR in progress.

I updated the code as you suggested me above.

Updated existing tests add added new tests: one for checkbox and an other for `always_remember_me` (It seems to work)

sadikoff commented:
Hey @bechir 

Sorry for so late ping a lot of time has passed, but will you be able to update your PR and do some last pending fixes?

Cheers!

bechir commented:
Hi @sadikoff 
I rebased my work and tests passed.
Can you check it please?

weaverryan commented:
Sorry for how slow this was to get merged, but thank you for your EXCELLENT work on this feature - very good stuff @bechir!
